assets/src/dashboard/components/typeahead/index.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { KEYS } from '../../constants';
+import { Close as CloseIcon, Search as SearchIcon } from '../../icons';
+import useFocusOut from '../../utils/useFocusOut';
+import TypeaheadOptions from './typeaheadOptions';
+import useTypeahead from './useTypeahead';
+import {
+  SearchContainer,
+  InputContainer,
+  ControlVisibilityContainer,
+  StyledInput,
+  SearchButton,
+  ClearInputButton,
+} from './components';
+
+const Typeahead = ({
+  inputId,
+  items,
+  className,
+  disabled,
+  onChange,
+  placeholder,
+  value = '',
+  ariaLabel,
+  isFiltering,
+  ...rest
+}) => {
+  const {
+    isMenuOpen,
+    isInputExpanded,
+    showMenu,
+    menuFocused,
+    inputValue,
+    selectedValueIndex,
+  } = useTypeahead({ items, value });
+
+  const searchRef = useRef();
+  const inputRef = useRef();
+  const typeaheadOptionsRef = useRef();
+
+  const focusInput = useCallback(() => {
+    inputRef.current.focus();
+  }, [inputRef]);
+
+  const handleFocusOut = useCallback(() => {
+    showMenu.set(false);
+    menuFocused.set(false);
+  }, [showMenu, menuFocused]);
+
+  useFocusOut(searchRef, handleFocusOut);
+
+  const handleInputChange = useCallback(
+    (item) => {
+      if (!showMenu.value) {
+        showMenu.set(true);
+      }
+      inputValue.set(item.label);
+      onChange(item.value.trim());
+    },
+    [onChange, inputValue, showMenu]
+  );
+
+  const handleMenuItemSelect = useCallback(
+    (item) => {
+      if (!item.value) {
+        return;
+      }
+      handleInputChange(item);
+      showMenu.set(false);
+    },
+    [handleInputChange, showMenu]
+  );
+
+  const handleClearInputButtonFocus = useCallback(() => {
+    showMenu.set(false);
+  }, [showMenu]);
+
+  const handleInputClear = useCallback(() => {
+    handleInputChange({ label: '', value: '' });
+    showMenu.set(false);
+    menuFocused.set(false);
+  }, [handleInputChange, menuFocused, showMenu]);
+
+  const handleKeyDown = useCallback(
+    (event) => {
+      if (event.key === KEYS.DOWN) {
+        event.preventDefault();
+        typeaheadOptionsRef.current?.children[selectedValueIndex.value].focus();
```

Comment:
So I think this is where it is happening

<img width="340" alt="Screen Shot 2020-09-30 at 9 00 21 PM" src="https://user-images.githubusercontent.com/1738349/94757509-0b85e480-0360-11eb-97ed-f236951be91a.png">

I cannot arrow down here


<img width="458" alt="Screen Shot 2020-09-30 at 9 00 02 PM" src="https://user-images.githubusercontent.com/1738349/94757505-07f25d80-0360-11eb-954f-1fdfb8e45e67.png">

```
const handleKeyDown = Object(react__WEBPACK_IMPORTED_MODULE_0__["useCallback"])(event=>{
                if (event.key === _constants__WEBPACK_IMPORTED_MODULE_3__["KEYS"].DOWN) {
                    var _typeaheadOptionsRef$;

                    event.preventDefault();
                    (_typeaheadOptionsRef$ = typeaheadOptionsRef.current) === null || _typeaheadOptionsRef$ === void 0 ? void 0 : _typeaheadOptionsRef$.children[selectedValueIndex.value].focus();
                }
            }
```

Source map aside, this is where it is crashing.

Reply:
<img width="495" alt="Screen Shot 2020-09-30 at 9 06 34 PM" src="https://user-images.githubusercontent.com/1738349/94757754-d62dc680-0360-11eb-81aa-47d99226eda8.png">

On further investigation it looks like the ref is being assigned to an old style ref and the value to `-1`.

Reply:
thanks for doing detective work on this before passing it back, this is a really helpful deduction.